File: common/src/main/java/juuxel/adorn/block/SneakClickHandler.java
package juuxel.adorn.block;

import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.InteractionResult;
import net.minecraft.world.InteractionHand;
import net.minecraft.world.phys.BlockHitResult;
import net.minecraft.core.BlockPos;
import net.minecraft.world.level.Level;

/**
 * Implement on blocks to give them a sneak-click action.
 */
public interface SneakClickHandler {
    InteractionResult onSneakClick(BlockState state, Level world, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hitResult);
}

File: src/main/kotlin/juuxel/adorn/api/block/SneakClickHandler.kt
package juuxel.adorn.api.block

import net.minecraft.block.BlockState
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.util.ActionResult
import net.minecraft.util.Hand
import net.minecraft.util.hit.BlockHitResult
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World

/**
 * Implement on blocks to give them a sneak-click action.
 */
interface SneakClickHandler {
    fun onSneakClick(state: BlockState, world: World, pos: BlockPos, player: PlayerEntity, hand: Hand, hitResult: BlockHitResult): ActionResult
}
